community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/pipes/IdSeekIterator.scala

```diff
@@ -49,9 +49,7 @@ abstract class IdSeekIterator[T <: PropertyContainer]
   private def computeNextEntity(): T = {
     while (entityIds.hasNext) {
       val id = asLongEntityId(entityIds.next())
-      if (operations.exists(id.longValue()))
-        return operations.getById(id.longValue())
-
+      return operations.getByIdIfExists(id.longValue()).getOrElse(null.asInstanceOf[T])
```

Comment:
Perhaps a logic error here. Previously we only returned if the entity existed, otherwise moved on to the next one. Now we return even if it does not exist, and return null. To match older behaviour perhaps we should have something like:

```
   val maybeEntity = operations.getByIdIfExists(id.longValue())
   if(maybeEntity.isDefined) return maybeEntity.get
```